### PR TITLE
feat: create genrobots tool

### PR DESCRIPTION
**Pull request by tbpg**
This tool generates a robots.txt file based on the content in the
given buckets.

### CODE REVIEW

File: genrobots/main.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	"regexp"
+	"sort"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+)
+
+var (
+	versionRE = regexp.MustCompile(`(rev\d+-)?v?(\d+\.\d+\.\d+|HEAD)`)
```
Comment (codyoss) [inline genrobots/main.go:34]: nit: I find regex hard to read, might be nice to have a comment on this describing what it is doing.

File: genrobots/main.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+func main() {
+	googleapisBucketName := flag.String("gbucket", "", "googleapis.dev bucket name")
+	cgcBucketName := flag.String("cgcbucket", "", "c.g.c. bucket name")
+	robotsFilename := "robots.txt"
+
+	flag.Parse()
+
+	if *googleapisBucketName == "" {
+		fmt.Fprintf(os.Stderr, "must set -gbucket")
```
Comment (codyoss) [inline genrobots/main.go:45]: nit: seems like you are also using log, maybe that should be does for these first couple fmt calls as well. i.e: `log.Fatal(...)`

**Review by fhinkel — APPROVED** [review]

File: genrobots/main.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+		os.Exit(1)
+	}
+
+	client, err := storage.NewClient(context.Background())
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to create storage client: %v", err)
+		os.Exit(1)
+	}
+
+	ctx := context.Background()
```
Comment (codyoss) [inline genrobots/main.go:59]: This can be defied above L53 and passed to client.

File: genrobots/main.go [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+	if *googleapisBucketName == "" {
+		fmt.Fprintf(os.Stderr, "must set -gbucket")
+		os.Exit(1)
+	}
+	if *cgcBucketName == "" {
+		fmt.Fprintf(os.Stderr, "must set -cgcbucket")
+		os.Exit(1)
+	}
+
+	client, err := storage.NewClient(context.Background())
```
Comment (codyoss) [inline genrobots/main.go:53]: Not a big deal right now, but incase this code ever gets refactored might be a good idea to defer client.Close()

**Review by codyoss — APPROVED** [review]
Just a couple of nits or else LGTM!

File: genrobots/main.go (hunk 1 above)
Comment (tbpg) [reply genrobots/main.go]: Done.

File: genrobots/main.go (hunk 2 above)
Comment (tbpg) [reply genrobots/main.go]: Done.

File: genrobots/main.go (hunk 4 above)
Comment (tbpg) [reply genrobots/main.go]: Done. :man_facepalming:

File: genrobots/main.go (hunk 3 above)
Comment (tbpg) [reply genrobots/main.go]: Done.

**Review by tbpg — COMMENTED** [review]
Thank you!

**Pull request merged**